pkg/envoy/ads/stream.go

```diff
@@ -148,8 +151,9 @@ func shouldPushUpdate(proxy *envoy.Proxy) bool {
 	// In ADS, CDS and LDS will come first in all cases. Only allow an control-plane-push update push if
 	// we have sent either to the proxy already.
 	if proxy.GetLastSentNonce(envoy.TypeLDS) == "" && proxy.GetLastSentNonce(envoy.TypeCDS) == "" {
-		log.Error().Msgf("Proxy %s: LDS and CDS unrequested yet, waiting for first request for this proxy to be responded to",
-			proxy.String())
+		log.Error().Str(errcode.Kind, errcode.ErrProxyNotReady.String()).
```

Comment:
`ErrProxyNotReady` seems ambiguous here. Could you change this to `ErrUnexpectedXDSRequest` and update the description for the same.

Reply:
Would `ErrProxyWaitingForInitialUpdate` work?

Reply:
Update to `ErrUnexpectedXDSRequest`